test/recipes/90-test_shlibload.t

```diff
@@ -18,16 +18,23 @@ BEGIN {
 use lib bldtop_dir('.');
 use configdata;
 
+# When libssl and libcrypto are compiled on Linux with "-rpath", but not
+# "--enable-new-dtags", the RPATH takes precedence over LD_LIBRARY_PATH,
+# and we end up running with the wrong libraries.  This is resolved by
+# using full paths to the shared objects.
+#
+my $top = bldtop_dir('.');
+
 plan skip_all => "Test only supported in a shared build" if disabled("shared");
 plan skip_all => "Test is disabled on AIX" if config('target') =~ m|^aix|;
 
 plan tests => 4;
 
 my $libcrypto_idx = $unified_info{rename}->{libcrypto} // "libcrypto";
 my $libssl_idx = $unified_info{rename}->{libssl} // "libssl";
-my $libcrypto =
+my $libcrypto = "$top/".
     $unified_info{sharednames}->{$libcrypto_idx}.$target{shared_extension_simple};
-my $libssl =
+my $libssl = "$top/".
     $unified_info{sharednames}->{$libssl_idx}.$target{shared_extension_simple};
```

Comment:
These are incorrect, the "simple" names are NOT derived from `$unified_info{sharednames}`.

It is really preferable if you compute the real SONAME / dll name, the same way as in `util/shlib_wrap.sh.in`

Reply:
@levitte Care to send a patch?  I am a bit out of my depth on build-system internals.  Just added a prefix to get to the right directory, which seems to work, but if there's a better way, I'm open to suggestions.

Reply:
Maybe you want to take over this PR, and I'll review/approve your version, and close this one?

Reply:
The `shlib_wrap.sh` inline Perl is your code, and it already the bulk of this PR...

Reply:
... yeahok

Reply:
So what shall it be? Will you submit a replacement?